src/react.js

```diff
@@ -1,21 +1,79 @@
+import TransisObject from "./object";
 import TransisArray from "./array";
+import {getPath} from "./util";
+
+let nextId = 1;
+let updateLog = {};
+let updateQueue = {};
+
+function componentCmp(a, b) {
+  if (a._transisId < b._transisId) { return -1; }
+  else if (a._transisId > b._transisId) { return 1; }
+  else { return 0; }
+}
+
+function preFlush() {
+  updateLog = {};
+  updateQueue = {};
+  TransisObject.delay(postFlush);
+}
+
+function postFlush() {
+  let components = [];
+
+  for (let id in updateQueue) {
+    components.push(updateQueue[id]);
+  }
+
+  // Sort the components by their assigned _transisId. Since components get mounted from the top
+  // down, this should ensure that parent components are force updated before any descendent
+  // components that also need an update. This avoids the case where we force update a component
+  // and then force update one of its ancestors, which may unnecessarily render the component
+  // again.
+  components.sort(componentCmp).forEach(function(component) {
+    if (!updateLog[component._transisId] && component.isMounted()) {
+      component.forceUpdate();
+    }
+  });
+
+  TransisObject.delayPreFlush(preFlush);
+}
+
+function queueUpdate(component) {
+  updateQueue[component._transisId] = component;
+}
+
+function logUpdate(component) {
+  updateLog[component._transisId] = true;
+}
+
+TransisObject.delayPreFlush(preFlush);
 
 export var PropsMixin = function(props) {
   return {
     componentWillMount: function() {
-      this._transisFU = this._transisFU || (() => { this.isMounted() && this.forceUpdate(); });
+      this._transisId = this._transisId || nextId++;
+      this._transisQueueUpdate = this._transisQueueUpdate || (() => { queueUpdate(this); });
```

Comment:
This would cause a problem if you included both mixins into the same component:

```
Uncaught Invariant Violation: ReactClassInterface: You are attempting to define `_doit` on your component more than once. This conflict may be due to a mixin.
```